lte/gateway/python/magma/pipelined/app/enforcement_stats.py

```diff
@@ -567,21 +603,93 @@ def _get_rule_id(self, flow):
                               rule_num, e)
             return ""
 
-def _generate_rule_match(imsi, ip_addr, rule_num, version, direction):
+    def _prepare_session_config_report(self, stats_msgs):
+        session_config_dict={}
+
+        for flow_stats in stats_msgs:
+            for stat in flow_stats:
+                if stat.table_id != self.tbl_num:
+                    return
+
+                local_f_teid_ng = _get_ng_local_f_id(stat)
+                if not local_f_teid_ng or local_f_teid_ng == REG_ZERO_VAL:
+                    continue
+                #Already present
+                if local_f_teid_ng in session_config_dict:
+                    continue
+
+                sid = _get_sid(stat)
+                if not sid:
+                    continue
+
+                rule_version = _get_version(stat)
+                session_config_dict.update ({local_f_teid_ng:
+                                             UPFSessionState(subscriber_id=sid,
+                                                             session_version=rule_version,
+                                                             local_f_teid=local_f_teid_ng)})
+
+        #Send Session messages
+        SessionStateManager.report_session_config_state(session_config_dict,
+                                                        self.ng_config.sessiond_setinterface)
+
+    def deactivate_rules(self, imsi, ip_addr, rule_ids):
+        """
+        Deactivate flows for a subscriber.
+            imsi+ipv4+rule_ids -> remove rules for specific imsi session
+
+        Args:
+            imsi (string): subscriber id
+            ip_addr (string): subscriber ip address
+            rule_ids (list of strings): policy rule ids
+        """
+        if not self.init_finished:
+            self.logger.error('Pipelined is not initialized')
+            return RuleModResult.FAILURE
+
+        for rule_id in rule_ids:
+            self._deactivate_flow_for_rule(imsi, ip_addr, rule_id)
+
+    def _deactivate_flow_for_rule(self, imsi, ip_addr, rule_id):
```

Comment:
this is essentially _delete_flow

Reply:
Not required for it. Removed the method.